File: src/MixNetFx/FacFunctions/Dialogs/EmployeeInfoRelated/frmEmployeePicker.vb
Imports Facesso.Data
Imports System.Windows.Forms

Public Class frmEmployeePicker

    Sub New()

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Me.AcceptButton = ucEmployeePicker.btnOK
        Me.CancelButton = ucEmployeePicker.btnCancel
        AddHandler ucEmployeePicker.btnOK.Click, AddressOf OKClickHandler
        AddHandler ucEmployeePicker.btnCancel.Click, AddressOf CancelClickHandler

    End Sub

    Public Function GetEmployeeSelection() As EmployeeInfoItems
        Using Me
            ucEmployeePicker.Employees = New EmployeeInfoItems(0)
            Me.ShowDialog()
            If Me.DialogResult = Windows.Forms.DialogResult.OK Then
                Return ucEmployeePicker.SelectedEmployees
            Else
                Return Nothing
            End If
        End Using
    End Function

    Private Sub OKClickHandler(ByVal sender As Object, ByVal e As EventArgs)
        Me.DialogResult = Windows.Forms.DialogResult.OK
    End Sub

    Private Sub CancelClickHandler(ByVal sender As Object, ByVal e As EventArgs)
        Me.DialogResult = Windows.Forms.DialogResult.Cancel
    End Sub

End Class

File: src/NetFxCSharp/Facesso.Functions/Dialogs/EmployeeInfoRelated/frmEmployeePicker.cs
using Facesso.Data;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.Functions
{
    public partial class frmEmployeePicker
    {
        public frmEmployeePicker()
        {
            // This call is required by the Windows Form Designer.
            InitializeComponent();
            // Add any initialization after the InitializeComponent() call.
            this.AcceptButton = ucEmployeePicker.btnOK;
            this.CancelButton = ucEmployeePicker.btnCancel;
            ucEmployeePicker.btnOK.Click += OKClickHandler;
            ucEmployeePicker.btnCancel.Click += CancelClickHandler;
        }

        public EmployeeInfoItems GetEmployeeSelection()
        {
            using (this)
            {
                ucEmployeePicker.Employees = new EmployeeInfoItems(0);
                this.ShowDialog();
                if (this.DialogResult == System.Windows.Forms.DialogResult.OK)
                {
                    return ucEmployeePicker.SelectedEmployees;
                }
                else
                {
                    return null;
                }
            }

            return default(EmployeeInfoItems);
        }

        private void OKClickHandler(object sender, EventArgs e)
        {
            this.DialogResult = System.Windows.Forms.DialogResult.OK;
        }

        private void CancelClickHandler(object sender, EventArgs e)
        {
            this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
        }
    }
}